src/Raven.Client/Documents/Changes/DatabaseChanges.cs

```diff
@@ -528,6 +528,9 @@ private async Task DoWork(string nodeTag)
                 catch (OperationCanceledException e)
                 {
                     NotifyAboutError(e);
+                    if (wasConnected)
```

Comment:
Why bother checking if `wasConnected`? A status changed should be safe to always invoke, no?

Reply:
The only place when `_cts` is cancelled is on dispose of changes, not sure if we need to execute event then. Is there another situation when we can get that exception? If yes, then maybe if the `_cts` is cancelled we should not invoke it and return, but if this is different source we should continue to reconnect?

By looking at description of the issue that you wrote, you are trying to solve reconnecting by using that function, but the bug is elsewhere IMO.

Reply:
`By looking at description of the issue that you wrote, you are trying to solve reconnecting by using that function, but the bug is elsewhere IMO.`

That is basically why I started the discussion. I have ran into a situation where the changes connection has failed and it is no longer attempting to reconnect. I did not get an error notification on my subscription (most likely the error occurred after the call to changes api and before my Rx subscription get registered). I want to know whether the database changes is in this state.  If there is another call I can make to detect this, then I can use that. 
An alternative solution may be to either expose the task created by DoWork or to have a method that returns the state of the DoWork task instead of using the ensure connected now.

Reply:
`Why bother checking if wasConnected? A status changed should be safe to always invoke, no?`

I was just following the same pattern as in the other exception handler. I assume it was done to avoid sending a connection status event if there was not change in the status.